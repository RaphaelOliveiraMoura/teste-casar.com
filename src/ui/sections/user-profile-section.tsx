import Image from "next/image";

import { User } from "@/domain/entities/user";

type UserProfileSectionProps = {
  user: User;
};

export function UserProfileSection({ user }: UserProfileSectionProps) {
  return (
    <aside className="sticky top-4 flex flex-col items-center gap-2 border p-6 text-center">
      <Image
        src={user.imageUrl}
        alt={`Imagem de perfil do usuÃ¡rio ${user.name}`}
        width={200}
        height={200}
        className="rounded-full bg-grey-neutral"
      />
      <div>
        <h2 className="text-xl font-semibold">{user.name}</h2>
        <span>@{user.id}</span>
      </div>
      <p>{user.description}</p>
    </aside>
  );
}
