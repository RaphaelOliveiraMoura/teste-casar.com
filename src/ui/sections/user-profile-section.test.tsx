import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";

import { UserProfileSection } from "./user-profile-section";

test("<UserProfileSection />", async () => {
  const snapshot = render(
    <UserProfileSection
      user={{
        id: "1",
        name: "user test",
        description: "user description",
        imageUrl: "https://google.com",
      }}
    />,
  );

  expect(screen.getByText("user test")).toBeDefined();
  expect(screen.getByText("user description")).toBeDefined();

  expect(snapshot).toMatchSnapshot();
});
