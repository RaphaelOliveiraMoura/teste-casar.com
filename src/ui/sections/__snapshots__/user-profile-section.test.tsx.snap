// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserProfileSection /> 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="sticky top-4 flex flex-col items-center gap-2 border p-6 text-center"
      >
        <img
          alt="Imagem de perfil do usuário user test"
          class="rounded-full bg-grey-neutral"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fgoogle.com&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fgoogle.com&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fgoogle.com&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
        <div>
          <h2
            class="text-xl font-semibold"
          >
            user test
          </h2>
          <span>
            @
            1
          </span>
        </div>
        <p>
          user description
        </p>
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="sticky top-4 flex flex-col items-center gap-2 border p-6 text-center"
    >
      <img
        alt="Imagem de perfil do usuário user test"
        class="rounded-full bg-grey-neutral"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fgoogle.com&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgoogle.com&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fgoogle.com&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
      <div>
        <h2
          class="text-xl font-semibold"
        >
          user test
        </h2>
        <span>
          @
          1
        </span>
      </div>
      <p>
        user description
      </p>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
